compiler-core/src/config.rs

```diff
@@ -62,63 +59,94 @@ impl<'de> Deserialize<'de> for SpdxLicense {
     }
 }
 
+impl Serialize for SpdxLicense {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.licence)
+    }
+}
+
 impl AsRef<str> for SpdxLicense {
     fn as_ref(&self) -> &str {
         self.licence.as_str()
     }
 }
 
-#[derive(Deserialize, Debug, PartialEq, Clone)]
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
 pub struct PackageConfig {
-    #[serde(with = "package_name")]
+    #[serde(deserialize_with = "package_name::deserialize")]
     pub name: EcoString,
     #[serde(default = "default_version")]
     pub version: Version,
     #[serde(
         default,
-        rename = "gleam",
-        serialize_with = "serialise_range",
-        deserialize_with = "deserialise_range"
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "serialise_hex_range",
+        deserialize_with = "deserialise_hex_range"
     )]
+    // gleam version field in Hex format (for [de]serialization purposes)
+    pub gleam: Option<version::Range>,
```

Comment:
This should be part of the `gleam_version` field rather than a second one. They are tightly coupled as being two representations of the same thing, so it is invalid to have one without the other.